tasks/prime.py

```diff
@@ -260,6 +261,31 @@ def update_mto_shipment_address(self):
         if success:
             self.replace_prime_data(PrimeObjects.MTO_SHIPMENT, mto_shipment, new_shipment)
 
+    @tag(PrimeObjects.MTO_AGENT.value, "updateMTOAgent")
+    @task
+    def update_mto_agent(self):
+        mto_shipment = self.get_random_data(
+            PrimeObjects.MTO_SHIPMENT
+        )  # this grabs response payload from the PrimeObjects.MTO_AGENT list when load testing runs
+        if not mto_shipment:
+            return  # can't run this task
+
+        payload = self.fake_request("/mto-shipments/{mtoShipmentID}/agents/{agentID}", "put")
+        mto_agent = mto_shipment["agents"][0]
+        headers = {"content-type": "application/json", "If-Match": mto_agent["eTag"]}
+        resp = self.client.put(
+            prime_path(f"mto-shipments/{mto_shipment['id']}/agents/{mto_agent['id']}"),
+            name=prime_path("mto-shipments/:mtoShipmentID/agents/mtoAgentID"),
+            data=json.dumps(payload),
+            headers=headers,
+            **self.user.cert_kwargs,
+        )
+
+        resp, success = check_response(resp, "Update MTO Agent", payload)
+
+        if success:
+            self.replace_prime_data(PrimeObjects.MTO_AGENT, mto_agent, resp)
```

Comment:
After pondering this, I think we should be setting a new MTO_SHIPMENT and not an agent here. The reason is because we grab a shipment to start with anyway, so we're never going to use these agents. I'd suggest doing what Chris did for `update_mto_shipment_address`:

```python
if success:
    new_shipment = deepcopy(mto_shipment)
    new_shipment["agents"][0] = resp
    self.replace_prime_data(PrimeObjects.MTO_SHIPMENT, mto_shipment, new_shipment)
```

Let me know if you have any questions about this and we can talk about it in more detail!

Reply:
Nice! Thanks for the feedback. I see a HUGE improvement and no failures so far now when running the tests.